### PR TITLE
[#152040583] Transactions

### PR DESCRIPTION
**Pull request by guillaumedebavelaere**

### CODE REVIEW

File: TradeItIosEmsApi/TradeItIosEmsApiLib.h [hunk 1]
```diff
@@ -88,6 +88,11 @@
 #import "TradeItPriceInfo.h"
 #import "TradeItCancelOrderRequest.h"
 
+// Transactions
+#import "TradeItTransactionsHistoryRequest.h"
+#import "TradeItTransactionsHistoryResult.h"
+#import "TradeItTransactionHistory.h"
```
Comment (jsom) [inline TradeItIosEmsApi/TradeItIosEmsApiLib.h]: I think this should just be called `TradeItTransaction`

File: TradeItIosEmsApi/TradeItTransactionHistory.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+@property (nonatomic, copy, nullable) NSNumber<Optional>  * amount;
+
+@property (nonatomic, copy, nullable) NSString<Optional>  * action;
+
+@property (nonatomic, copy, nullable) NSString<Optional>  * type;
+
+@property (nonatomic, copy, nullable) NSString<Optional>  * id;
+
+@property (nonatomic, copy, nullable) NSNumber<Optional>  * quantity;
```
Comment (jsom) [inline TradeItIosEmsApi/TradeItTransactionHistory.h]: Usually it's written `NSNumber<Optional> *quantity`

File: TradeItIosEmsApi/TradeItTransactionsHistoryResult.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+@class TradeItTransactionHistory;
+
+@protocol TradeItTransactionHistory
+
+@end
+
+@interface TradeItTransactionsHistoryResult : TradeItResult
+
+@property(nonatomic, copy, nullable) NSArray<TradeItTransactionHistory*> <Optional, TradeItTransactionHistory>  * transactionHistoryDetailsList;
```
Comment (jsom) [inline TradeItIosEmsApi/TradeItTransactionsHistoryResult.h]: Is the array actually nullable? Why?

File: TradeItIosTicketSDK2/TradeItTransactionService.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    
+    init(session: TradeItSession) {
+        self.session = session
+    }
+    
+    func getTransactionsHistory(
+        _ data: TradeItTransactionsHistoryRequest,
+        onSuccess: @escaping (TradeItTransactionsHistoryResult) -> Void,
+        onFailure: @escaping (TradeItErrorResult) -> Void
+        ) {
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTransactionService.swift]: Indentation is off

File: TradeItIosEmsApi/TradeItTransactionsHistoryRequest.h [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#import "TradeItRequest.h"
+
+@interface TradeItTransactionsHistoryRequest : TradeItRequest
+
+    @property (nonatomic, copy, nullable) NSString * token;
+    @property (nonatomic, copy, nullable) NSString * accountNumber;
```
Comment (jsom) [inline TradeItIosEmsApi/TradeItTransactionsHistoryRequest.h]: Remove indentation

File: TradeItIosEmsApi/TradeItIosEmsApiLib.h (hunk 1 above)
Comment (jsom) [reply TradeItIosEmsApi/TradeItIosEmsApiLib.h]: Have these been added to both the SDK and SDK-Carthage targets?

Comment (guillaumedebavelaere) [reply TradeItIosEmsApi/TradeItIosEmsApiLib.h]: > I think this should just be called TradeItTransaction

Ok, I will rename it

> Have these been added to both the SDK and SDK-Carthage targets?

Yes

File: TradeItIosEmsApi/TradeItTransactionsHistoryResult.h (hunk 3 above)
Comment (guillaumedebavelaere) [reply TradeItIosEmsApi/TradeItTransactionsHistoryResult.h]: No, but I put all optional to avoid blowing up in case the server returns null

Comment (jsom) [reply TradeItIosEmsApi/TradeItTransactionsHistoryResult.h]: The server should not return `null` here ever. Do we have an E2E test that asserts this returns an array?

Comment (jsom) [reply TradeItIosEmsApi/TradeItTransactionsHistoryResult.h]: @mitochondrion thoughts?

**Comment by guillaumedebavelaere** [comment]
@jsom @mitochondrion is it good to merge ?

**Pull request merged**